components/main/Desktop/ExportButton.jsx

```diff
@@ -43,20 +49,110 @@ function ExportButton({ filters }) {
+    const startYear = moment(filters.startDate).year();
+    const endYear = moment(filters.endDate).year();
+
+    const getAllRequests = (year, startDate, endDate, councilId = '', status = '') => `
+      SELECT * FROM requests_${year} 
+      WHERE CreatedDate >= '${startDate}' 
+      AND CreatedDate < '${endDate}'
+      ${status !== '' ? ` AND Status='${status}'` : ''}
+      ${councilId !== null ? ` AND NC='${councilId}'` : ''} 
+      AND RequestType IN (${formattedRequestTypes})`;
+
+    // Note: this logic will only generate the SR count CSV if it meets the following conditions:
+    // exactly one SR type is selected, a NC is selected, and status is Open.
+    const groupRequestsByAddress = (year, startDate, endDate, councilId) => `
+      SELECT Address, COUNT(*) AS NumberOfRequests FROM requests_${year}
+      WHERE CreatedDate >= '${startDate}' 
+      AND CreatedDate < '${endDate}' 
+      AND Status = 'Open' 
+      AND NC = '${councilId}' 
+      AND RequestType IN (${formattedRequestTypes})
+      GROUP BY Address`;
+
+    const generateQuery = (grouped = false) => {
+      if (startYear === endYear) {
+        if (grouped) {
+          // SRs grouped by address from same year
+          return groupRequestsByAddress(
+            startYear,
+            filters.startDate,
+            filters.endDate,
+            filters.councilId,
+          );
+        }
+        // data comes from same year and includes all columns matching filters
+        return getAllRequests(
+          startYear,
+          filters.startDate,
+          filters.endDate,
+          filters.councilId,
+          requestStatusFilter,
+        );
+      }
+
+      const endOfStartYear = moment(filters.startDate).endOf('year').format('YYYY-MM-DD');
+      const startOfEndYear = moment(filters.endDate).startOf('year').format('YYYY-MM-DD');
+
+      // SRs grouped by address with different start and end years
+      if (grouped) {
+        return `(${groupRequestsByAddress(
+          startYear,
+          filters.startDate,
+          endOfStartYear,
+          filters.councilId,
+        )}) UNION ALL (${groupRequestsByAddress(
+          endYear,
+          startOfEndYear,
+          filters.endDate,
+          filters.councilId,
+        )})`;
+      }
+
+      // data with different start and end years and includes all columns matching filters
+      return `(${getAllRequests(
+        startYear,
+        filters.startDate,
+        endOfStartYear,
+        filters.councilId,
+        requestStatusFilter,
+      )}) UNION ALL (${getAllRequests(
+        endYear,
+        startOfEndYear,
+        filters.endDate,
+        filters.councilId,
+        requestStatusFilter,
+      )})`;
+    };
+
+    const neighborhoodDataQuery = generateQuery();
+    const neighborhoodDataToExport = await conn.query(neighborhoodDataQuery);
+    const neighborhoodResults = ddbh.getTableData(neighborhoodDataToExport);
+
```

Comment:
I would do the timestamp-to-date conversion here before passing neighborhoodResults to `Papa.unparse`.

Something like
````javascript
const formattedResults = neighborhoodResults.map(row => ({
  ...row,
  CreatedDate: row.CreatedDate ? moment(row.CreatedDate).format('YYYY-MM-DD HH:mm:ss') : null,
  UpdatedDate: row.UpdatedDate ? moment(row.UpdatedDate).format('YYYY-MM-DD HH:mm:ss') : null,
  serviceDate: row.serviceDate ? moment(row.serviceDate).format('YYYY-MM-DD HH:mm:ss') : null,
}));
````

Reply:
Need to include ClosedDate column in there